src/backend/distributed/planner/deparse_shard_query.c

```diff
@@ -350,6 +365,34 @@ UpdateRelationsToLocalShardTables(Node *node, List *relationShardList)
 }
 
 
+/*
+ * FindRelationShardByOid finds the RelationShard with given Oid.
+ * Search for the restrictions associated with the RTE. There better be
+ * some, otherwise this query wouldn't be eligible as a router query.
+ *
+ * FIXME: We should probably use a hashtable here, to do efficient
+ * lookup.
+ */
+RelationShard *
+FindRelationShardByOid(Oid oid, List *relationShardList)
+{
+	ListCell *relationShardCell = NULL;
```

Comment:
minor: could keep it right before the loop to tell that it's just used to iterate